### PR TITLE
Improved documentation

### PR DESCRIPTION
**Pull request by nicolas-van**
Hello,

This is a big change in the organization of the documentation. The story is that I lost quite a lot of time trying to understand how this project works and had to read directly the code in the end because I wasn't able to understand the current doc.

I kept most of it, re-organized it in a more logical way, provided a full working example and modernized a little the examples that already existed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,380 +1,414 @@
...
+  storageOptions: {
+    sequelize: sequelize
+  }
+})
 
-  // The type of the column holding migration name.
-  // Defaults to `Sequelize.STRING`
-  columnType: new Sequelize.STRING(100)
-}
+;(async () => {
```
Comment (sushantdhiman) [inline README.md]: unnecessary semi

Comment (nicolas-van) [reply README.md]: Actually no, it is strictly mandatory. You can even test it to see it won't compile.

That's just a common pattern found in no-semicolon javascript coding style to force a proper understanding by the compiler.

Comment (sushantdhiman) [reply README.md]: Right, codebase is not using semi

**Comment by papb** [comment]
This fantastic, thank you very much! Excellent!!

**Comment by nicolas-van** [comment]
Thanks :)

**Pull request merged**